statsmodels/base/transform.py

```diff
@@ -0,0 +1,220 @@
+import numpy as np
+from statsmodels.robust import mad
+from scipy.optimize import minimize_scalar
+from scipy.stats import boxcox_normmax
+
+
+class BoxCox(object):
+    """
+    Mixin class to allow for a Box-Cox transformation.
+    """
+
+    def transform_boxcox(self, x, lmbda=None, method='guerrero', **kwargs):
+        """
+        Performs a Box-Cox transformation on the data array x. If lmbda is None,
+        the indicated method is used to estimate a suitable lambda parameter.
+
+        Parameters
+        ----------
+        lmbda : float
+            The lambda parameter for the Box-Cox transform. If None, a value
+            will be estimated by means of the specified method.
+        method : {'guerrero', 'loglik'}
+            The method to estimate the lambda parameter. Will only be used if
+            lmbda is None, and defaults to 'guerrero', detailed in Guerrero
+            (1993). 'loglik' maximizes the profile likelihood.
+        **kwargs: dict
+            Options for the specified method.
+            * For 'guerrero', this entails _R_, the grouping parameter,
+              and _scale_, the dispersion measure.
+            * For 'loglik', this dict is not used.
+
+        Returns
+        -------
+        y : array_like
+            The transformed series.
+        lmbda : float
+            The lmbda parameter used to transform the series.
+
+        References
+        ----------
+        Guerrero, Victor M. 1993. "Time-series analysis supported by power
+        transformations". `Journal of Forecasting`. 12 (1): 37-48.
+
+        Guerrero, Victor M. and Perera, Rafael. 2004. "Variance Stabilizing
+        Power Transformation for Time Series," `Journal of Modern Applied
+        Statistical Methods`. 3 (2): 357-369.
+
+        Box, G. E. P., and D. R. Cox. 1964. "An Analysis of Transformations".
+        `Journal of the Royal Statistical Society`. 26 (2): 211-252.
+        """
+        x = np.asarray(x)
+
+        if np.any(x <= 0):
+            raise ValueError("Non-positive x.")
+
+        if lmbda is None:
+            lmbda = self._est_lambda(x,
+                                     bounds=(-1, 2),
+                                     method=method,
+                                     **kwargs)
+
+        # if less than 0.01, treat lambda as zero.
+        if np.abs(lmbda) < 10e-3:
+            y = np.log(x)
+        else:
+            y = (np.power(x, lmbda) - 1.) / lmbda
```

Comment:
In #2892 pr I've already changed the code to yours, and the difference lambda handler between transform and untransform caused problem.
For example, if the input lambda is 0.001 which less than 0.01, the transform will treat it as np.log, but untransform used boxcox with 0.001 correctly, then the endog is not the same as untransformed data.